server/src/main/java/org/opensearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -205,6 +207,7 @@ boolean tryCollectFromTermFrequencies(LeafReaderContext ctx, SortedSetDocValues
 
         TermsEnum indexTermsEnum = segmentTerms.iterator();
         BytesRef indexTerm = indexTermsEnum.next();
+        final SortedSetDocValues globalOrds = this.getGlobalOrds(ctx);
```

Comment:
Aha so it was loading the globalOrds even before checking whether it is an EFFECTIVE match all query ( L182-L194 )